### PR TITLE
Extend AggregationHandle and its subclasses to handle DISTINCT aggregation

### PR DESCRIPTION
**Pull request by jianqiao**
This PR adds virtual methods in `AggregationHandle` and corresponding implements in its subclasses to handle DISTINCT aggregations.

### CODE REVIEW

File: expressions/aggregation/AggregationConcreteHandle.hpp [hunk 1]
```diff
@@ -59,6 +62,17 @@ class AggregationConcreteHandle : public AggregationHandle {
                << "takes at least one argument.";
   }
 
+  AggregationStateHashTableBase* createDistinctifyHashTable(
```
Comment (pateljm) [inline expressions/aggregation/AggregationConcreteHandle.hpp]: Add comments to make it easier to follow. Same for the notes below. Brief comments are ok.

File: expressions/aggregation/AggregationHandleAvg.hpp [hunk 2]
```diff
@@ -143,6 +143,13 @@ class AggregationHandleAvg : public AggregationConcreteHandle {
       const AggregationStateHashTableBase &hash_table,
       std::vector<std::vector<TypedValue>> *group_by_keys) const override;
 
+  AggregationState* aggregateOnDistinctifyHashTableForSingle(
```
Comment (pateljm) [inline expressions/aggregation/AggregationHandleAvg.hpp]: Doxygen style comments here would be nice. Thanks!

File: expressions/aggregation/AggregationHandleCount.hpp [hunk 3]
```diff
@@ -151,6 +151,13 @@ class AggregationHandleCount : public AggregationConcreteHandle {
       const AggregationStateHashTableBase &hash_table,
       std::vector<std::vector<TypedValue>> *group_by_keys) const override;
 
+  AggregationState* aggregateOnDistinctifyHashTableForSingle(
```
Comment (pateljm) [inline expressions/aggregation/AggregationHandleCount.hpp]: Doxygen comments?

File: expressions/aggregation/AggregationHandleDistinct.hpp [hunk 4]
```diff
@@ -80,6 +80,19 @@ class AggregationHandleDistinct : public AggregationConcreteHandle {
     LOG(FATAL) << "AggregationHandleDistinct does not support finalize().";
   }
 
+  AggregationState* aggregateOnDistinctifyHashTableForSingle(
```
Comment (pateljm) [inline expressions/aggregation/AggregationHandleDistinct.hpp]: Doxygen comments?

File: expressions/aggregation/AggregationHandleMax.hpp [hunk 5]
```diff
@@ -135,6 +135,13 @@ class AggregationHandleMax : public AggregationConcreteHandle {
       const AggregationStateHashTableBase &hash_table,
       std::vector<std::vector<TypedValue>> *group_by_keys) const override;
 
+  AggregationState* aggregateOnDistinctifyHashTableForSingle(
```
Comment (pateljm) [inline expressions/aggregation/AggregationHandleMax.hpp]: Doxygen comments.

File: expressions/aggregation/AggregationHandleMin.hpp [hunk 6]
```diff
@@ -134,6 +134,13 @@ class AggregationHandleMin : public AggregationConcreteHandle {
       const AggregationStateHashTableBase &hash_table,
       std::vector<std::vector<TypedValue>> *group_by_keys) const override;
 
+  AggregationState* aggregateOnDistinctifyHashTableForSingle(
```
Comment (pateljm) [inline expressions/aggregation/AggregationHandleMin.hpp]: Doxygen comments here and the next function.

Comment (jianqiao) [inline expressions/aggregation/AggregationHandleMin.hpp]: It is the implementation of `AggregationHandle::aggregateOnDistinctifyHashTableForSingle()`.
Perhaps just brief note here 
`See AggregationHandle::aggregateOnDistinctifyHashTableForSingle() for documentation`?

**Comment by jianqiao** [comment]
@pateljm Just added more documentation. Please see if that helps read the code.

**Comment by pateljm** [comment]
LTGM. Merging.

**Pull request merged**